### DISCUSSION TITLE
Jdownloader Rename Rule Integration

### DISCUSSION THREAD

**Discussion by MorBidWon** [aclap-dev/video-downloadhelper#1316] [Help and Q&A]
### Your question

I use Video DownloadHelper to copy the link to clipboard and it transfer over to Jdownloader2 with no problem, but it gives a strange file name (and always the same).   How do you setup a package rule to rename it in the linkgrabber?

In the Video helper box it shows the file name I would like but when it copies over to Jdownloader it becomes something like:  index-f3-va-a1 (1080p).mp4   all of the llinks have the same name.  In the web page URL the name is there:  /xx/x/NAME/  and it is also displayed correctly in download helper just above the copy URL button.

### Addon & CoApp version and details

Version 9.1.0.44 (stable), target mozilla, locale en-US.

Companion App installed (v2.0.19)

**Comment by Mattmjs** [comment]
Why do you need another software, just do the download in VDH itself and you have the right file name.

**Reply by MorBidWon** [reply]
Because it is slower (500k/s vs 1,500k) and seem to be only able to do 5 at a time, with Jdownload I can have 10 at time at the higher speed.

**Reply by GrampaWildWilly** [reply]
I do not work for VDH.  I'm just another user, same as Matt.

People periodically come in here complaining about download speed.  But if you're going to complain about it, you're going to have to share some URLs so somebody else can see what speed he can get.  I have seen VDH download at over 30 million bytes per second.  The limiting factors are (a) the bandwidth you are buying from your ISP, & (b) throttling by the serving web site.  If you're going to claim there's a bottleneck on download speed, you need to show it happening in the Windows Resource Monitor.  I'm assuming you're using Windows.  You failed to mention that detail in your problem description.

As for concurrency, there is a setting in VDH for that.  Have you tried looking through the VDH settings for something that might be relevant to this subject?

When it comes to integration with some other product, I don't think that should be a design goal of VDH.  VDH is its own product & should operate for the convenience of itself.  If you want to use some other downloader, don't expect VDH to adapt to it, nor the other product to adapt to VDH.
